pyresample/geometry.py

```diff
@@ -1131,6 +1131,9 @@ def compute_domain(
             x_resolution, y_resolution = resolution
             width = int(np.rint((corners[2] - corners[0]) * 1.0 / x_resolution + 1))
             height = int(np.rint((corners[3] - corners[1]) * 1.0 / y_resolution + 1))
+            # align corners with pixel resolution
+            corners[2] = corners[0] + (width - 1) * x_resolution
+            corners[3] = corners[1] + (height - 1) * y_resolution
```

Comment:
doesn't this rather belong to `_update_corners_for_full_extent`?

Reply:
Not strictly, but I agree it seems like an odd place to put this code compared to the existing version of the code (updating corners when only width/height or x/y resolution were calculated). I don't think it goes in `_update_corners_for_full_extents` since that is mostly concerned with areas that are the full extent of the projection and only deals with `x` extents at the moment.

Reply:
Also note the conversion in the related issue that may require other changes to this PR.